tests/fast_tests/test_environments.py

```diff
@@ -0,0 +1,383 @@
+                                net_params=net_params)
+
+        self.env = TestEnv(env_params, sumo_params, scenario)
+
+    def tearDown(self):
+        self.env.terminate()
+        self.env = None
+
+    def test_obs_space(self):
+        self.assertEqual(self.env.observation_space.shape[0], 0)
+        self.assertEqual(len(self.env.observation_space.high), 0)
+        self.assertEqual(len(self.env.observation_space.low), 0)
+
+    def test_action_space(self):
+        self.assertEqual(self.env.action_space.shape[0], 0)
+        self.assertEqual(len(self.env.action_space.high), 0)
+        self.assertEqual(len(self.env.action_space.low), 0)
+
+    def test_get_state(self):
+        self.assertEqual(len(self.env.get_state()), 0)
+
+    def test_compute_reward(self):
+        # test the default
+        self.assertEqual(self.env.compute_reward([], []), 0)
+
+        # test if the "reward_fn" parameter is defined
+        def reward_fn(*_):
+            return 1
+        self.env.env_params.additional_params["reward_fn"] = reward_fn
+        self.assertEqual(self.env.compute_reward([], []), 1)
+
+
+class TestAccelEnv(unittest.TestCase):
+
+    """Tests the AccelEnv environment in flow/envs/loop/loop_accel.py"""
+
+    def setUp(self):
+        vehicles = Vehicles()
+        vehicles.add("rl", acceleration_controller=(RLController, {}))
+        vehicles.add("human", acceleration_controller=(IDMController, {}))
+
+        net_params = NetParams(additional_params=LOOP_PARAMS)
+        env_params = EnvParams(additional_params=ACCELENV_PARAMS)
+        sumo_params = SumoParams()
+
+        scenario = LoopScenario("test_loop",
+                                generator_class=CircleGenerator,
+                                vehicles=vehicles,
+                                net_params=net_params)
+
+        self.env = AccelEnv(env_params, sumo_params, scenario)
+
+    def tearDown(self):
+        self.env.terminate()
+        self.env = None
+
+    def test_observed_ids(self):
+        self.env.additional_command()
+        self.assertListEqual(self.env.vehicles.get_observed_ids(),
+                             self.env.vehicles.get_human_ids())
+
+    def test_action_space(self):
+        self.assertEqual(self.env.action_space.shape[0],
+                         self.env.vehicles.num_rl_vehicles)
+        self.assertEqual(self.env.action_space.high,
+                         self.env.env_params.additional_params["max_accel"])
+        self.assertEqual(self.env.action_space.low,
+                         -self.env.env_params.additional_params["max_decel"])
+
+    def test_get_state(self):
+        expected_state = np.array([[self.env.vehicles.get_speed(veh_id)
+                                    / self.env.scenario.max_speed,
+                                    self.env.get_x_by_id(veh_id) /
+                                    self.env.scenario.length]
+                                   for veh_id in self.env.sorted_ids])
```

Comment:
I sort of agree with Nish, what is the point of tests like these? This is just copying code and checking that the code you've copied is the same. Note, this is meant constructively; I'm sure I'm missing something.

Reply:
they're for two reasons:
- solidifying certain aspects of the codebase that we are not interested in developing on anymore
- maintaining some level of confidence that an environment will train. since we cannot easily run RL tests to completion, we can at least ensure that the environments are still the same (I have lost a good deal of time trying to figure out what's changed in an old environment just to find out that I or someone else had introduced new bugs, probably when they were trying something locally)

generally speaking, tests for environments should cover more intricate methods, but the ones I tested were pretty generic, minus I guess testing the `reset` method for `WaveAttenuationEnv`

Reply:
Okay, I'm convinced of the value of this for now. We should replace these later with some sort of regression testing (TBD).

Reply:
So in conclusion, it's a saved version of the envs.

Reply:
yup, I was thinking the same about regression testing. these could just be a sanity test for now